src/cwe_checker_lib/src/analysis/pointer_inference/context/trait_impls.rs

```diff
@@ -32,6 +32,36 @@ impl<'a> crate::analysis::forward_interprocedural_fixpoint::Context<'a> for Cont
             };
             let _ = self.log_collector.send(LogThreadMsg::Cwe(warning));
         }
+        // check for out-of-bounds memory access
+        if state.contains_out_of_bounds_mem_access(&def.term, self.runtime_memory_image) {
+            let (warning_name, warning_description) = match def.term {
+                Def::Load { .. } => (
+                    "CWE125",
+                    format!(
+                        "(Out-of-bounds Read) Memory load at {} may be out of bounds",
+                        def.tid.address
+                    ),
+                ),
+                Def::Store { .. } => (
+                    "CWE787",
+                    format!(
+                        "(Out-of-bounds Write) Memory write at {} may be out of bounds",
+                        def.tid.address
+                    ),
+                ),
+                Def::Assign { .. } => panic!(),
```

Comment:
Is is on purpose that you panic without a message?

Reply:
Since `state.contains_out_of_bounds_mem_access(...)` always returns `false` for `Def::Assign`, the panic will never execute. So I was too lazy to think of an error message for it.